src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -670,4 +677,117 @@ export default class GeGiftEntryFormApp extends NavigationMixin(LightningElement
+                .then(closeModalCallback)
+                .catch((error) =>
+                    handleError(error)
+                );
+        } else {
+            closeModalCallback();
+        }
+    }
+
+    @wire(getObjectInfo, { objectApiName: DATA_IMPORT_BATCH_OBJECT})
+    dataImportBatchObjectInfo;
+
+    get userCanEditBatchTableColumns() {
+        try {
+            return this.dataImportBatchObjectInfo.data.fields[
+                BATCH_TABLE_COLUMNS_FIELD.fieldApiName
+                ].updateable;
+        } catch (e) {
+            return false;
+        }
+    }
+
+    handleSelectColumns(event) {
+        const modalContent =
+            this.userCanEditBatchTableColumns ?
+                this.buildModalConfigSelectColumns(
+                    event.detail.options,
+                    event.detail.values) :
+                this.modalConfigNoAccess;
+
+        registerListener('privateselectcolumns', this.handleReceiveEvent, this);
+
+        this.dispatchEvent(new CustomEvent('togglemodal', {
+            detail: modalContent
+        }));
+    }
+
+    buildModalConfigSelectColumns(available, selected) {
+        const modalHeader = this.CUSTOM_LABELS.geTabBatchTableColumns;
+        const modalConfig = {
+            componentProperties: {
+                name: 'selectcolumnsmodal',
+                cssClass: 'slds-m-bottom_medium slds-p-horizontal_small',
+                sourceLabel: this.CUSTOM_LABELS.geLabelCustomTableSourceFields,
+                selectedLabel: this.CUSTOM_LABELS.geLabelCustomTableSelectedFields,
+                dedicatedListenerEventName: 'privateselectcolumns',
+                options: available,
+                values: selected,
+                showModalFooter: true,
+                min: 1
+            },
+            modalProperties: {
+                cssClass: 'slds-modal_large',
+                header: modalHeader,
+                componentName: 'utilDualListbox',
+                showCloseButton: true
+            }
+        };
+        return modalConfig;
+    }
+
+    get modalConfigNoAccess() {
+        const modalHeader = this.CUSTOM_LABELS.geTabBatchTableColumns;
+        const errorFieldName =
+            `${BATCH_TABLE_COLUMNS_FIELD.objectApiName}: (${BATCH_TABLE_COLUMNS_FIELD.fieldApiName})`;
+        const errorFLSBody = GeLabelService.format(
+            this.CUSTOM_LABELS.geErrorFLSBody, [errorFieldName]
+        );
+        const message = `${this.CUSTOM_LABELS.geErrorFLSBatchTableColumns}  ${errorFLSBody}`;
+        const modalConfig = {
+            componentProperties: {
+                name: 'noaccessmodal',
+                illustrationClass: 'slds-p-around_large',
+                dedicatedListenerEventName: 'privateselectcolumns',
+                showModalFooter: false,
+                message: message,
+                title: this.CUSTOM_LABELS.geErrorFLSHeader,
+                variant: 'lake-mountain'
```

Comment:
I believe we're now using a new illustration Daniel added `no-access` for the various crud/fls messaging.
```suggestion
                variant: 'no-access'
```

Reply:
👍🏽